Update v1.5.x docs with latest changes (#139)
NordicSemiconductor/asset-tracker-cloud-docs: docs/azure/ContinuousIntegration.rst

   The setup process in Azure is more complicated when compared to the :ref:`AWS continuous integration setup <aws-continuous-integration>`, since it involves many manual steps, which cannot be automated.
   If you have ideas to simplify the process, `please provide your input <https://github.com/NordicSemiconductor/asset-tracker-cloud-azure-js/issues/1>`_.

> Login and create the resource group

Reviewer: ```suggestion
Log in and create the resource group
```

===================================

To create the resource group for the CI resources, complete the following steps:

> 1. Login to the shell:

Reviewer: ```suggestion
1. Log in to the shell:
```

   .. code-block:: bash

       az login

#. Export the identifier of the subscription which contains the nRF Asset Tracker resources:

   .. parsed-literal::
      :class: highlight

      export SUBSCRIPTION_ID="*subscription id*"

#. Make sure that you have enabled the right subscription by using the following commands:

   .. code-block:: bash

       az account set --subscription $SUBSCRIPTION_ID 
       # Verify that it is set to default
       az account list --output table

#. Choose a resource group name for the solution and export it as ``RESOURCE_GROUP``.

>    In this example, we use ``nrfassettrackerci`` as the resource group name.

Reviewer: ```suggestion
   In this example, ``nrfassettrackerci`` is used as the resource group name.
```

   .. code-block:: bash

      export RESOURCE_GROUP="nrfassettrackerci"
    
#. Choose a name for the solution and export it as ``APP_NAME``.

>    Use a short name (not more than 16 characters) composed of numbers and lower-case letters only.

Reviewer 2: ```suggestion
   Use a short name (not more than 16 characters) composed of numbers and lowercase letters only.
```

>    In this example, we use ``nrfassettrackerci`` as the application name.

Reviewer: ```suggestion
   In this example, ``nrfassettrackerci`` is used as the application name.
```

   .. parsed-literal::
      :class: highlight

      export APP_NAME="nrfassettrackerci"

#. Choose a resource group and a name for the Device Update instance and export it as ``ADU_RESOURCE_GROUP`` and ``ADU_INSTANCE_NAME``.

>    There is an undocumented limit of 2 instances per ADU account, so the CI should be run against a separate Device Update instance.

Reviewer: ```suggestion
   There is an undocumented limit of two instances per ADU account, so the CI should be run against a separate Device Update instance.
```

>    There is an undocumented limit of 2 instances per ADU account, so the CI should be run against a separate Device Update instance.

Reviewer 2: ```suggestion
   There is an undocumented limit of two instances per ADU account, so the CI must be run against a separate Device Update instance.
```

>    In this example, we use ``nRFAssetTrackerADUCI`` as the resource group and Device Update instance name.

Reviewer: ```suggestion
   In this example, ``nRFAssetTrackerADUCI`` is used as the resource group and Device Update instance name.
```

   .. code-block:: bash

      # add to .envrc
      export ADU_RESOURCE_GROUP="nRFAssetTrackerADUCI"
      export ADU_INSTANCE_NAME="nRFAssetTrackerADUCI"

#. Configure your preferred location (you can list the locations using ``az account list-locations``) and export it on the environment variable ``LOCATION``.

>    In this example, we use ``northeurope`` as the location name.

Reviewer: ```suggestion
   In this example, ``northeurope`` is used as the location name.
```

   #. Follow the instructions in the `tutorial for setting up a resource owner password credentials flow in Azure Active Directory B2C <https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-ropc-policy?tabs=app-reg-ga&pivots=b2c-user-flow#register-an-application>`_ and register an application.
      Make sure to:

      - set the :guilabel:`Supported account types` to :guilabel:`All users`     
      - update the manifest and set
        
        .. code-block:: json

           {"oauth2AllowImplicitFlow": true}
   
#. Export the initial domain name that you used:

   .. parsed-literal::
      :class: highlight

      export B2C_TENANT="*Initial domain name*" # For example, "nrfassettrackerciusers"

#. Export the :guilabel:`Application (client) ID` and the :guilabel:`Directory (tenant) ID` of the created Active Directory B2C App:

   .. parsed-literal::
      :class: highlight

      export APP_REG_CLIENT_ID="*Application (client) id*"
      export B2C_TENANT_ID="*Directory (tenant) ID*"

> #. For enabling the test-runner to programmatically login users, enable `the resource owner password credentials (ROPC) flow <https://docs.microsoft.com/EN-US/azure/active-directory-b2c/configure-ropc?tabs=app-reg-ga>`_ with the following settings on the Active Directory B2C:

Reviewer: ```suggestion
#. For enabling the test-runner to programmatically log in users, enable `the resource owner password credentials (ROPC) flow <https://docs.microsoft.com/EN-US/azure/active-directory-b2c/configure-ropc?tabs=app-reg-ga>`_ with the following settings on the Active Directory B2C:
```

   
   #. Click :guilabel:`+ Add a scope` and create a new scope with the following values and click :guilabel:`Add a scope`:
      
      * Scope name - ``nrfassettracker.admin``
      * Admin consent display name - Admin access to the nRF Asset Tracker API
      * Admin consent description - Allows admin access to all resources exposed through the nRF Asset Tracker API

   #. Click :guilabel:`API permissions` and then click :guilabel:`+ Add a permission`. Under :guilabel:`My APIs`, select the app registration.
   
   #. Enable the ``nrfassettracker.admin`` permission and click :guilabel:`Add permission`.
   
#. Click :guilabel:`Grant admin consent for <your B2C directory>`.
   
#. Create a new client secret for the App registration (for example, ``12OzW72ie-U.vlmzik-eO5gX.x26jLTI6U``) and note it down.

   .. parsed-literal::
      :class: highlight

      export B2C_CLIENT_SECRET="*client secret*"

#. Link this Azure AD B2C tenant to the subscription for CI by following the `Billing guide <https://docs.microsoft.com/en-us/azure/active-directory-b2c/billing#link-an-azure-ad-b2c-tenant-to-a-subscription>`_.

> Setup continuous integration on GitHub

Reviewer: ```suggestion
Set up continuous integration on GitHub
```
